### ISSUE TITLE
MobileNetV2 and AlexNet can't be exported to ONNX correctly

### ISSUE THREAD

**Issue by FedyuninV** [pytorch/pytorch#35644]
## 🐛 Bug

When given the same input the differences between net results is too big between PyTorch and onnxruntime.

## To Reproduce

```python
import torch
import torch.nn as nn
import torchvision
import onnxruntime
import numpy as np

torch.manual_seed(17)
np.random.seed(17)

# channelwise conv
image_height = 224
image_width = 224
image_channels = 3
batch_size = 10

np_dummy_input = np.array(list(range(batch_size * image_width * image_height * image_channels)), dtype=np.float32)
# channel last -> channel first
np_dummy_input = np_dummy_input.reshape((batch_size, image_height, image_width, image_channels)).transpose((0, 3, 1, 2))
dummy_input = torch.from_numpy(np_dummy_input)

model = torchvision.models.mobilenet_v2(pretrained=True).cpu()

with torch.no_grad():
    flatten_res = model(dummy_input).numpy().flatten()
    print(flatten_res)

torch.onnx.export(model, dummy_input, "temp.onnx", verbose=False)

ort_session = onnxruntime.InferenceSession("temp.onnx")
ort_inputs = {ort_session.get_inputs()[0].name: np_dummy_input}
ort_outs = ort_session.run(None, ort_inputs)
print(np.array(ort_outs[0]).flatten())

# Example Output:
# [-0.687555  -3.6106596 -8.104299  ... -0.917107   5.467206  -2.1354108]
# [-0.40397647 -0.45896444 -1.148888   ... -2.288803    4.361473
#  -0.31830937]
```

The same goes for the AlexNet.

## Expected behavior

Only minor differences between pytorch's output and onnxruntime's.

## Environment

 - PyTorch Version (e.g., 1.0): 1.4
 - OS (e.g., Linux): Windows 10
 - How you installed PyTorch (`conda`, `pip`, source): `conda`
 - Build command you used (if compiling from source):
 - Python version: 3.6.10
 - CUDA/cuDNN version: (not used anyway)
 - GPU models and configuration: (not used anyway)
 - Any other relevant information:

## Additional context

First: why *pretrained* networks have *significantly* different outputs on the same input with different seeds and `with torch.no_grad()` mode?

After some experiments it seems that the problem appears to be in BatchNorm2d layer:

```python
torch.manual_seed(1234)
np.random.seed(1234)

# conv
image_height = 224
image_width = 224
image_channels = 32
batch_size = 5
filter_count = 32

np_dummy_input = np.array(list(range(batch_size * image_width * image_height * image_channels)), dtype=np.float32)
np_dummy_input = np_dummy_input.reshape((batch_size, image_height, image_width, image_channels)).transpose((0, 3, 1, 2))
dummy_input = torch.from_numpy(np_dummy_input)

class Net(nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        self.conv = nn.Conv2d(image_channels, filter_count, (3, 3), stride=1, padding=1,
                             groups=1, bias=True)
        self.bn = nn.BatchNorm2d(filter_count, eps=1e-3, momentum=0.1)
        self.conv1 = nn.Conv2d(filter_count, filter_count, (3, 3), stride=1, padding=1,
                             groups=1, bias=True)
        self.bn1 = nn.BatchNorm2d(filter_count, eps=1e-3, momentum=0.1)
        self.conv2 = nn.Conv2d(filter_count, filter_count, (3, 3), stride=1, padding=1,
                             groups=1, bias=True)
        self.bn2 = nn.BatchNorm2d(filter_count, eps=1e-3, momentum=0.1)
        self.conv3 = nn.Conv2d(filter_count, filter_count, (3, 3), stride=1, padding=1,
                             groups=1, bias=True)
        self.bn3 = nn.BatchNorm2d(filter_count, eps=1e-3, momentum=0.1)
    
    def forward(self, x):
        return self.bn3(self.conv3(self.bn2(self.conv2(self.bn1(self.conv1(self.bn(self.conv(x))))))))

model = Net().to("cpu")

with torch.no_grad():
    print(model(dummy_input).numpy().transpose((0, 2, 3, 1)).flatten())

torch.onnx.export(model, dummy_input, "temp.onnx", verbose=False, keep_initializers_as_inputs=True)

ort_session = onnxruntime.InferenceSession("temp.onnx")
ort_inputs = {ort_session.get_inputs()[0].name: np_dummy_input}
ort_outs = ort_session.run(None, ort_inputs)
print(np.array(ort_outs[0]).transpose((0, 2, 3, 1)).flatten())

# Example Output:
# [-1.2203052  -0.24037342 -0.5068347  ... -8.928441    1.8339034
#  -5.5446095 ]
# [-2.0485312e-02 -2.8994158e-02  2.0724810e-03 ... -2.8066349e+00
#  -1.3084781e+00 -2.0316699e+00]
```

Small nets with Conv or Gemm layers only work well.

**Comment by vadimkantorov** [comment]
you may want to put model explicitly in eval mode: `model.eval()` - `model.training` value affects BatchNorm behavior

**Comment by FedyuninV** [comment]
@vadimkantorov I didn't expect pretrained model to be in `train` state. I also didn't expect different behavior of BatchNorm in `no_grad` mode. My bad...

With `eval()` now all the nets have similar results with `onnxruntime`

Thanks a lot! Issue closed.
